File: models/consultation.go
// File: internal/models/consultation.go
package models

import "time"

type Consultation struct {
    ID               int       `json:"id" db:"id"`
    PatientID        int       `json:"patient_id"`
    DoctorID         int       `json:"doctor_id"`
    AppointmentID    int       `json:"appointment_id" db:"appointment_id"`
    ConsultationType string    `json:"consultation_type" db:"consultation_type"`
    Status           string    `json:"status" db:"status"`
    StartedAt        time.Time `json:"started_at" db:"started_at"`
    CompletedAt      time.Time `json:"completed_at" db:"completed_at"`
    Fee              float64   `json:"fee" db:"fee"`
}

// ConsultationFilter represents the filtering options for consultations
type ConsultationFilter struct {
    PatientID         int
    DoctorID          int
    AppointmentID     int
    ConsultationType  string
    Status            string
    StartDateFrom     time.Time
    StartDateTo       time.Time
    CompletedDateFrom time.Time
    CompletedDateTo   time.Time
}

File: src/lib/types/consultation.ts
export interface Consultation {
    id?: number;
    patient_id: number;
    doctor_id: number;
    status: 'requested' | 'in_progress' | 'completed' | 'cancelled';
    started_at?: string;
    completed_at?: string;
    fee: number;
    is_paid?: boolean;
    created_at?: string;
    updated_at?: string;
}

export interface ConsultationFilter {
    patient_id?: number;
    doctor_id?: number;
    status?: string;
    start_date_from?: string;
    start_date_to?: string;
    completed_date_from?: string;
    completed_date_to?: string;
}

// Align with the consultation_details table structure exactly
export interface ConsultationDetails {
    id?: number;
    consultation_id: number;
    request_details?: string;
    symptoms?: string;
    diagnosis?: string;
    prescription?: string;
    notes?: string;
    // Note: The database schema doesn't have follow_up_date, but we'll keep it for frontend use
    follow_up_date?: string;
    created_at?: string;
    updated_at?: string;
}